### DISCUSSION TITLE
how to add the alfen charger in the energy dashboard

### DISCUSSION THREAD

**Discussion by supermarcie** [leeyuentuen/alfen_wallbox#70] [Q&A]
What do I have to do to add the Alfen wallbox into the energy dashboard?

I found following : The wallbox can be added to the Home Assistant Energy Dashboard using the _meter_reading sensor.

but no idea where I need to add this in the dashboard.
Currently I configured different devices (electricity, solar, gas, water), but no idea where to add this "_meter_reading"?

**Comment by leeyuentuen** [comment]
I've add them as in "return to grid", not sure if this is the correct way.

at the beginning I though to put them in home battery storage, but for some reason (forgot why) it create a strange graph on it

<img width="1413" alt="image" src="https://github.com/leeyuentuen/alfen_wallbox/assets/1487966/69382f8c-f031-4020-a003-a91c9f6bcc14">

**Reply by aritmeester** [reply]
I also have configured the Alfen meter reading in the "return to grid" category. That way the costs vs revenue is also displayed correctly.
(We have an energy contract with dynamic pricing and a fixed rate for charging reimbursements)

![image](https://github.com/leeyuentuen/alfen_wallbox/assets/644662/bd59d667-a300-44fd-8f3a-4eef7e4ac0f7)

**Comment by citoki** [comment]
When the question was raised, it probably wasn't possible back then, but today I would recommend to add the charger `sensor.<your-wallbox>_meter_reading_socket_1 ` as individual device in <http://homeassistant.local:8123/config/energy/dashboard>
